contracts/ConvergentCurvePool.sol

```diff
@@ -140,7 +183,98 @@ contract ConvergentCurvePool is IMinimalSwapInfoPool, BalancerPoolToken {
         return _poolId;
     }
 
-    // Trade Functionality
+    ///////////////////////
+    /// Oracle support ///
+    /////////////////////
+
+    /// @notice Synchronize the bond and underlying balance and calculate the `_cumulativeBalancesRatio`.
+    function sync() external {
+        // Get pool tokens balances
+        (IERC20[] memory tokens, uint256[] memory balances, ) = _vault
+            .getPoolTokens(_poolId);
+        // It is assumed that the balances retrieved from the pool is always one trade before as funds in pool will only get
+        // updated after the successful swap/mint/burn.
+        (uint256 bondBalance, uint256 underlyingBalance) = address(tokens[0]) ==
+            address(underlying)
+            ? (balances[1], balances[0])
+            : (balances[0], balances[1]);
+        _sync(bondBalance, underlyingBalance);
+    }
+
+    /// @notice Synchronize the bond and underlying balance and calculate the `_cumulativeBalancesRatio`.
+    /// @param bondBalance Reserve of bond token.
+    /// @param underlyingBalance Reserve of underlying token.
+    function _sync(uint256 bondBalance, uint256 underlyingBalance) internal {
+        uint32 blockTimestamp = uint32(block.timestamp);
+        uint32 timeElapsed = blockTimestamp - _blockTimestampLast; // overflow is desired
+        if (timeElapsed > 0 && bondBalance != 0 && underlyingBalance != 0) {
+            // We multiply by 1e18 here so that r = t * y/x is a fixed point factor with 18 decimals
+            uint256 scaledBondBalance = (bondBalance + totalSupply()) * 1e18;
+            _cumulativeBalancesRatio += _toU224(
+                (scaledBondBalance * uint256(timeElapsed)) / underlyingBalance
+            );
+        }
+        emit Sync(
+            bondBalance,
+            underlyingBalance,
+            _blockTimestampLast = blockTimestamp
+        );
+    }
+
+    ///@notice Update the cumulative ratio and calculates the avg cumulative ratio for a given period.
+    function update() external {
```

Comment:
I don't think we need this function to be external. The only way the rate can change is during a swap so I don't see any need to arbitrarily update the rate oracle.  If nothing else...we need to talk this out.